packages/framework/src/Model/Product/Product.php

```diff
@@ -893,6 +905,109 @@ protected function setTranslations(ProductData $productData)
         }
     }
 
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductData $productData
+     */
+    private function setDomains($productData)
+    {
+        $domainIds = array_keys($productData->seoTitles);
+        foreach ($domainIds as $domainId) {
+            try {
+                $productDomain = $this->getProductDomain($domainId);
+            } catch (ProductDomainNotFoundException $e) {
+                $productDomain = new ProductDomain($this, $domainId);
+                $this->domains[$domainId] = $productDomain;
+            }
+            $productDomain->setSeoTitle($productData->seoTitles[$domainId]);
+            $productDomain->setSeoH1($productData->seoH1s[$domainId]);
+            $productDomain->setSeoMetaDescription($productData->seoMetaDescriptions[$domainId]);
+            $productDomain->setDescription($productData->descriptions[$domainId]);
+            $productDomain->setShortDescription($productData->shortDescriptions[$domainId]);
+        }
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Product\ProductDomain[]|\Doctrine\Common\Collections\ArrayCollection
+     */
+    public function getDomains()
```

Comment:
In Brand, method such as this wasn't public. The idea was that BrandDomain should look like an implementation detail from the outside.

Reply:
But, Brand has this method public, you need to have this public because of `ProductDataFactory` that needs these domains in `createFromBrand()` method, I thought of it and i dont see any other solution for not requiring these Domain entities

Reply:
How about iterating over `Domain::getAllIds()` in the factory and accessing then accessing the attributes via the main entity - eg. `Product::getSeoTitle($domainId)`?